docs/reference/migration/apis/data-stream-reindex.asciidoc

```diff
@@ -0,0 +1,358 @@
+[role="xpack"]
+[[data-stream-reindex-api]]
+=== Reindex data stream API
+++++
+<titleabbrev>Reindex data stream</titleabbrev>
+++++
+
+.New API reference
+[sidebar]
+--
+For the most up-to-date API details, refer to {api-es}/group/endpoint-migration[Migration APIs].
+--
+
+include::{es-ref-dir}/migration/apis/shared-migration-apis-tip.asciidoc[]
+
+The reindex data stream API is used to upgrade the backing indices of a data stream to the most
+recent major version. It works by reindexing each backing index then replacing the original
```

Comment:
maybe `It works by reindexing each backing index **into a new index,** then replacing the original`